+ source /root/GitRepo/mugen/libs/locallibs/common_lib.sh
++ python3 --version
++ '[' 0 -eq 0 ']'
++ source /root/GitRepo/mugen/libs/locallibs/common_lib_python.sh
+ main i
++ type -t post_test
+ '[' -n function ']'
+ trap post_test EXIT INT HUP TERM
+ rpm -qa
+ grep expect
++ type -t config_params
+ '[' -n '' ']'
++ type -t pre_test
+ '[' -n function ']'
+ pre_test
+ LOG_INFO 'Start environmental preparation.'
+ message='Start environmental preparation.'
+ python3 /root/GitRepo/mugen/libs/locallibs/mugen_log.py --level info --message 'Start environmental preparation.'
Wed Apr  5 07:00:05 2023 - INFO  - Start environmental preparation.
+ DNF_INSTALL 'cobbler httpd'
+ pkgs='cobbler httpd'
+ node=1
+ '[' -z '' ']'
+ tmpfile=
++ python3 /root/GitRepo/mugen/libs/locallibs/rpm_manage.py install --pkgs 'cobbler httpd' --node 1 --tempfile ''
+ tmpfile2=/tmp/tmp6blkhppc
+ '[' -z '' ']'
+ tmpfile=/tmp/tmp6blkhppc
+ systemctl start httpd
+ systemctl start cobblerd
+ cat /etc/yum.repos.d/openEuler.repo
+ LOG_INFO 'End of environmental preparation!'
+ message='End of environmental preparation!'
+ python3 /root/GitRepo/mugen/libs/locallibs/mugen_log.py --level info --message 'End of environmental preparation!'
Wed Apr  5 07:01:52 2023 - INFO  - End of environmental preparation!
++ type -t run_test
+ '[' -n function ']'
+ run_test
+ LOG_INFO 'Start to run testcase.'
+ message='Start to run testcase.'
+ python3 /root/GitRepo/mugen/libs/locallibs/mugen_log.py --level info --message 'Start to run testcase.'
Wed Apr  5 07:01:53 2023 - INFO  - Start to run testcase.
+ cobbler repo add --name=OpenEuler1 --mirror=/etc/yum.repos.d/openEuler_test.repo
cobblerd does not appear to be running/accessible: ConnectionRefusedError(111, 'Connection refused')
Traceback (most recent call last):
  File "/usr/bin/cobbler", line 35, in <module>
    sys.exit(app.main())
  File "/usr/lib/python3.10/site-packages/cobbler/cli.py", line 852, in main
    rc = cli.run(sys.argv)
  File "/usr/lib/python3.10/site-packages/cobbler/cli.py", line 418, in run
    self.token = self.remote.login("", self.shared_secret)
  File "/usr/lib64/python3.10/xmlrpc/client.py", line 1122, in __call__
    return self.__send(self.__name, args)
  File "/usr/lib64/python3.10/xmlrpc/client.py", line 1464, in __request
    response = self.__transport.request(
  File "/usr/lib64/python3.10/xmlrpc/client.py", line 1166, in request
    return self.single_request(host, handler, request_body, verbose)
  File "/usr/lib64/python3.10/xmlrpc/client.py", line 1196, in single_request
    raise ProtocolError(
xmlrpc.client.ProtocolError: <ProtocolError for 127.0.0.1:80/cobbler_api: 503 Service Unavailable>
+ CHECK_RESULT 1 0 0 'Failed option: repo add'
+ actual_result=1
+ expect_result=0
+ mode=0
+ error_log='Failed option: repo add'
+ exit_mode=0
+ '[' -z 1 ']'
+ '[' 0 -eq 0 ']'
+ test 1x '!=' 0x
+ test -n 'Failed option: repo add'
+ LOG_ERROR 'Failed option: repo add'
+ message='Failed option: repo add'
+ python3 /root/GitRepo/mugen/libs/locallibs/mugen_log.py --level error --message 'Failed option: repo add'
Wed Apr  5 07:01:59 2023 - ERROR - Failed option: repo add
+ (( exec_result++ ))
+ LOG_ERROR 'oe_test_cobbler_repo.sh line 34'
+ message='oe_test_cobbler_repo.sh line 34'
+ python3 /root/GitRepo/mugen/libs/locallibs/mugen_log.py --level error --message 'oe_test_cobbler_repo.sh line 34'
Wed Apr  5 07:02:00 2023 - ERROR - oe_test_cobbler_repo.sh line 34
+ '[' 0 -eq 1 ']'
+ return 0
+ cobbler repo report --name=OpenEuler1
+ grep 'Mirror.*: /etc/yum.repos.d/openEuler_test.repo'
+ CHECK_RESULT 1 0 0 'Failed option: repo add'
+ actual_result=1
+ expect_result=0
+ mode=0
+ error_log='Failed option: repo add'
+ exit_mode=0
+ '[' -z 1 ']'
+ '[' 0 -eq 0 ']'
+ test 1x '!=' 0x
+ test -n 'Failed option: repo add'
+ LOG_ERROR 'Failed option: repo add'
+ message='Failed option: repo add'
+ python3 /root/GitRepo/mugen/libs/locallibs/mugen_log.py --level error --message 'Failed option: repo add'
Wed Apr  5 07:02:05 2023 - ERROR - Failed option: repo add
+ (( exec_result++ ))
+ LOG_ERROR 'oe_test_cobbler_repo.sh line 36'
+ message='oe_test_cobbler_repo.sh line 36'
+ python3 /root/GitRepo/mugen/libs/locallibs/mugen_log.py --level error --message 'oe_test_cobbler_repo.sh line 36'
Wed Apr  5 07:02:06 2023 - ERROR - oe_test_cobbler_repo.sh line 36
+ '[' 0 -eq 1 ']'
+ return 0
+ cobbler repo copy --name=OpenEuler1 --newname=OpenEuler2
exception on server: 'internal error, unknown repo name OpenEuler1'
+ CHECK_RESULT 0 0 0 'Failed option: repo copy'
+ actual_result=0
+ expect_result=0
+ mode=0
+ error_log='Failed option: repo copy'
+ exit_mode=0
+ '[' -z 0 ']'
+ '[' 0 -eq 0 ']'
+ test 0x '!=' 0x
+ return 0
+ cobbler repo list
+ grep OpenEuler2
+ CHECK_RESULT 1 0 0 'Failed option: repo copy'
+ actual_result=1
+ expect_result=0
+ mode=0
+ error_log='Failed option: repo copy'
+ exit_mode=0
+ '[' -z 1 ']'
+ '[' 0 -eq 0 ']'
+ test 1x '!=' 0x
+ test -n 'Failed option: repo copy'
+ LOG_ERROR 'Failed option: repo copy'
+ message='Failed option: repo copy'
+ python3 /root/GitRepo/mugen/libs/locallibs/mugen_log.py --level error --message 'Failed option: repo copy'
Wed Apr  5 07:02:16 2023 - ERROR - Failed option: repo copy
+ (( exec_result++ ))
+ LOG_ERROR 'oe_test_cobbler_repo.sh line 40'
+ message='oe_test_cobbler_repo.sh line 40'
+ python3 /root/GitRepo/mugen/libs/locallibs/mugen_log.py --level error --message 'oe_test_cobbler_repo.sh line 40'
Wed Apr  5 07:02:17 2023 - ERROR - oe_test_cobbler_repo.sh line 40
+ '[' 0 -eq 1 ']'
+ return 0
+ cobbler repo edit --name=openEuler1 --owners=tom
exception on server: 'internal error, unknown repo name openEuler1'
+ CHECK_RESULT 0 0 0 'Failed option: repo edit'
+ actual_result=0
+ expect_result=0
+ mode=0
+ error_log='Failed option: repo edit'
+ exit_mode=0
+ '[' -z 0 ']'
+ '[' 0 -eq 0 ']'
+ test 0x '!=' 0x
+ return 0
+ cobbler repo report --name=openEuler1
+ grep tom
+ CHECK_RESULT 1 0 0 'Failed option: repo edit'
+ actual_result=1
+ expect_result=0
+ mode=0
+ error_log='Failed option: repo edit'
+ exit_mode=0
+ '[' -z 1 ']'
+ '[' 0 -eq 0 ']'
+ test 1x '!=' 0x
+ test -n 'Failed option: repo edit'
+ LOG_ERROR 'Failed option: repo edit'
+ message='Failed option: repo edit'
+ python3 /root/GitRepo/mugen/libs/locallibs/mugen_log.py --level error --message 'Failed option: repo edit'
Wed Apr  5 07:02:27 2023 - ERROR - Failed option: repo edit
+ (( exec_result++ ))
+ LOG_ERROR 'oe_test_cobbler_repo.sh line 44'
+ message='oe_test_cobbler_repo.sh line 44'
+ python3 /root/GitRepo/mugen/libs/locallibs/mugen_log.py --level error --message 'oe_test_cobbler_repo.sh line 44'
Wed Apr  5 07:02:28 2023 - ERROR - oe_test_cobbler_repo.sh line 44
+ '[' 0 -eq 1 ']'
+ return 0
+ cobbler repo list
+ grep OpenEuler
+ CHECK_RESULT 1 0 0 'Failed option: repo list'
+ actual_result=1
+ expect_result=0
+ mode=0
+ error_log='Failed option: repo list'
+ exit_mode=0
+ '[' -z 1 ']'
+ '[' 0 -eq 0 ']'
+ test 1x '!=' 0x
+ test -n 'Failed option: repo list'
+ LOG_ERROR 'Failed option: repo list'
+ message='Failed option: repo list'
+ python3 /root/GitRepo/mugen/libs/locallibs/mugen_log.py --level error --message 'Failed option: repo list'
Wed Apr  5 07:02:34 2023 - ERROR - Failed option: repo list
+ (( exec_result++ ))
+ LOG_ERROR 'oe_test_cobbler_repo.sh line 46'
+ message='oe_test_cobbler_repo.sh line 46'
+ python3 /root/GitRepo/mugen/libs/locallibs/mugen_log.py --level error --message 'oe_test_cobbler_repo.sh line 46'
Wed Apr  5 07:02:34 2023 - ERROR - oe_test_cobbler_repo.sh line 46
+ '[' 0 -eq 1 ']'
+ return 0
+ grep OpenEuler
+ cobbler repo find
+ CHECK_RESULT 1 0 0 'Failed option: repo find'
+ actual_result=1
+ expect_result=0
+ mode=0
+ error_log='Failed option: repo find'
+ exit_mode=0
+ '[' -z 1 ']'
+ '[' 0 -eq 0 ']'
+ test 1x '!=' 0x
+ test -n 'Failed option: repo find'
+ LOG_ERROR 'Failed option: repo find'
+ message='Failed option: repo find'
+ python3 /root/GitRepo/mugen/libs/locallibs/mugen_log.py --level error --message 'Failed option: repo find'
Wed Apr  5 07:02:40 2023 - ERROR - Failed option: repo find
+ (( exec_result++ ))
+ LOG_ERROR 'oe_test_cobbler_repo.sh line 48'
+ message='oe_test_cobbler_repo.sh line 48'
+ python3 /root/GitRepo/mugen/libs/locallibs/mugen_log.py --level error --message 'oe_test_cobbler_repo.sh line 48'
Wed Apr  5 07:02:41 2023 - ERROR - oe_test_cobbler_repo.sh line 48
+ '[' 0 -eq 1 ']'
+ return 0
+ cobbler repo remove --name=OpenEuler1
exception on server: 'internal error, unknown repo name OpenEuler1'
+ CHECK_RESULT 0 0 0 'Failed option: repo remove'
+ actual_result=0
+ expect_result=0
+ mode=0
+ error_log='Failed option: repo remove'
+ exit_mode=0
+ '[' -z 0 ']'
+ '[' 0 -eq 0 ']'
+ test 0x '!=' 0x
+ return 0
+ cobbler repo list
+ grep OpenEuler1
+ CHECK_RESULT 1 0 1 'Failed option: repo remove'
+ actual_result=1
+ expect_result=0
+ mode=1
+ error_log='Failed option: repo remove'
+ exit_mode=0
+ '[' -z 1 ']'
+ '[' 1 -eq 0 ']'
+ test 1x == 0x
+ return 0
+ cobbler repo rename --name=OpenEuler2 --newname=OpenEuler3
exception on server: 'internal error, unknown repo name OpenEuler2'
+ CHECK_RESULT 0 0 0 'Failed option: repo rename'
+ actual_result=0
+ expect_result=0
+ mode=0
+ error_log='Failed option: repo rename'
+ exit_mode=0
+ '[' -z 0 ']'
+ '[' 0 -eq 0 ']'
+ test 0x '!=' 0x
+ return 0
+ cobbler repo list
+ grep OpenEuler3
+ CHECK_RESULT 1 0 0 'Failed option: repo rename'
+ actual_result=1
+ expect_result=0
+ mode=0
+ error_log='Failed option: repo rename'
+ exit_mode=0
+ '[' -z 1 ']'
+ '[' 0 -eq 0 ']'
+ test 1x '!=' 0x
+ test -n 'Failed option: repo rename'
+ LOG_ERROR 'Failed option: repo rename'
+ message='Failed option: repo rename'
+ python3 /root/GitRepo/mugen/libs/locallibs/mugen_log.py --level error --message 'Failed option: repo rename'
Wed Apr  5 07:03:00 2023 - ERROR - Failed option: repo rename
+ (( exec_result++ ))
+ LOG_ERROR 'oe_test_cobbler_repo.sh line 56'
+ message='oe_test_cobbler_repo.sh line 56'
+ python3 /root/GitRepo/mugen/libs/locallibs/mugen_log.py --level error --message 'oe_test_cobbler_repo.sh line 56'
Wed Apr  5 07:03:01 2023 - ERROR - oe_test_cobbler_repo.sh line 56
+ '[' 0 -eq 1 ']'
+ return 0
+ grep Name
+ cobbler repo report --name=OpenEuler3
+ CHECK_RESULT 1 0 0 'Failed option: repo report'
+ actual_result=1
+ expect_result=0
+ mode=0
+ error_log='Failed option: repo report'
+ exit_mode=0
+ '[' -z 1 ']'
+ '[' 0 -eq 0 ']'
+ test 1x '!=' 0x
+ test -n 'Failed option: repo report'
+ LOG_ERROR 'Failed option: repo report'
+ message='Failed option: repo report'
+ python3 /root/GitRepo/mugen/libs/locallibs/mugen_log.py --level error --message 'Failed option: repo report'
Wed Apr  5 07:03:07 2023 - ERROR - Failed option: repo report
+ (( exec_result++ ))
+ LOG_ERROR 'oe_test_cobbler_repo.sh line 58'
+ message='oe_test_cobbler_repo.sh line 58'
+ python3 /root/GitRepo/mugen/libs/locallibs/mugen_log.py --level error --message 'oe_test_cobbler_repo.sh line 58'
Wed Apr  5 07:03:07 2023 - ERROR - oe_test_cobbler_repo.sh line 58
+ '[' 0 -eq 1 ']'
+ return 0
+ LOG_INFO 'End to run testcase.'
+ message='End to run testcase.'
+ python3 /root/GitRepo/mugen/libs/locallibs/mugen_log.py --level info --message 'End to run testcase.'
Wed Apr  5 07:03:08 2023 - INFO  - End to run testcase.
+ CASE_RESULT 0
+ case_re=0
+ test -z 8
+ test 8 -gt 0
+ LOG_ERROR 'failed to execute the case.'
+ message='failed to execute the case.'
+ python3 /root/GitRepo/mugen/libs/locallibs/mugen_log.py --level error --message 'failed to execute the case.'
Wed Apr  5 07:03:09 2023 - ERROR - failed to execute the case.
+ exit 8
+ post_test
+ LOG_INFO 'Start to restore the test environment.'
+ message='Start to restore the test environment.'
+ python3 /root/GitRepo/mugen/libs/locallibs/mugen_log.py --level info --message 'Start to restore the test environment.'
Wed Apr  5 07:03:10 2023 - INFO  - Start to restore the test environment.
+ systemctl stop httpd
+ systemctl stop cobblerd
+ rm -rf /etc/yum.repos.d/openEuler_test.repo
+ DNF_REMOVE
+ node=1
+ pkg_list=
+ mode=0
+ [[ -z /tmp/tmp6blkhppc ]]
+ '[' 0 -ne 0 ']'
+ '[' 1 == 0 ']'
+ python3 /root/GitRepo/mugen/libs/locallibs/rpm_manage.py remove --node 1 --pkgs '' --tempfile /tmp/tmp6blkhppc
+ '[' 0 -ne 0 ']'
+ LOG_INFO 'End to restore the test environment.'
+ message='End to restore the test environment.'
+ python3 /root/GitRepo/mugen/libs/locallibs/mugen_log.py --level info --message 'End to restore the test environment.'
Wed Apr  5 07:03:59 2023 - INFO  - End to restore the test environment.
